pkg/services/searchV2/extract/dashboard.go

```diff
@@ -139,6 +227,35 @@ func ReadDashboard(stream io.Reader, lookup DatasourceLookup) (*DashboardInfo, e
 		}
 	}
 
+	for i, panel := range dash.Panels {
+		var dsVariableRefs []DataSourceRef
+		var dsRefs []DataSourceRef
+
+		for i := range panel.Datasource {
+			isVariableRef := strings.HasPrefix(panel.Datasource[i].UID, "$")
+			if isVariableRef {
+				dsVariableRefs = append(dsVariableRefs, panel.Datasource[i])
+			} else {
+				dsRefs = append(dsRefs, panel.Datasource[i])
+			}
+		}
+
+		var referencedDs []DataSourceRef
+		for _, dsVariableRef := range dsVariableRefs {
+			variableName := dsVariableRef.UID
+			if strings.HasPrefix(variableName, "${") {
+				variableName = strings.TrimPrefix(strings.TrimSuffix(variableName, "}"), "${")
+			} else {
+				variableName = strings.TrimPrefix(variableName, "$")
+			}
+
+			refs := datasourceVariablesLookup.getDatasourceRefs(variableName)
+			referencedDs = append(referencedDs, refs...)
+		}
+
+		dash.Panels[i].Datasource = append(dsRefs, referencedDs...)
```

Comment:
Is this the same thing?
```suggestion
		for _, dsVariableRef := range dsVariableRefs {
			variableName := dsVariableRef.UID
			if strings.HasPrefix(variableName, "${") {
				variableName = strings.TrimPrefix(strings.TrimSuffix(variableName, "}"), "${")
			} else {
				variableName = strings.TrimPrefix(variableName, "$")
			}

			refs := datasourceVariablesLookup.getDatasourceRefs(variableName)
			dsRefs = append(dsRefs, refs...)
		}

		dash.Panels[i].Datasource = dsRefs
```

Reply:
I'd hope so, but right now the `Panel` model `Datasource` prop is an array, so I guess some of them could be variables and some of them could be proper references. I can't think of a scenario in which thats possible though

<img width="864" alt="image" src="https://user-images.githubusercontent.com/23451458/176542414-7578a351-bd72-406f-b4a2-325f77b7b665.png">